app/assets/javascripts/components/dashboard/create_task.jsx

```diff
@@ -6,38 +6,37 @@ class TaskCreationForm extends DefaultModal {
 
   constructor(props) {
     super();
-    this.state = {
-    };
+
     this.select = this.select.bind(this);
-    this.InputField = this.InputField.bind(this);
+    this.update = this.update.bind(this);
   }
 
-  InputField({ id, label, help, ...props }) {
-    const { Checkbox, Radio, FormGroup, ControlLabel, FormControl, Button, HelpBlock, TextArea } = ReactBootstrap;
-    return (
-      <FormGroup controlId={id} onChange={this.select}>
-        <ControlLabel>{label}</ControlLabel>
-        <br/>
-        <FormControl {...props} />
-        {help && <HelpBlock>{help}</HelpBlock>}
-      </FormGroup>
-    );
-  }
 
   select = (event) => {
     this.setState({[event.target.name]: event.target.value});
   }
 
+  update = (name, value) => {
+    this.setState({[name]: value});
+  }
+
   submit = () => {
     let clientInput = parseInt(_client_input.getCleanedMentionedUsers()[0]);
     let userInput = parseInt(_user_input.getCleanedMentionedUsers()[0]);
-    const payload = {
+    const task_params = {
       client_id: clientInput,
-      user_id: userInput,
       description: this.state.description,
       due_date: this.state.dueDate,
       title: this.state.name
     };
+    const user_params = {
+      user_id: userInput
+    }
+    const payload = {
+      task: task_params,
```

Comment:
You can also nest the dictionary without creating new variables like `task_params` and `user_params`. Like `{task: {client_id: clientInput, ...}, user: {user_id: userInput}}`. And, if you are just sending `user_id` for user params, then it might be better to not nest it, and have it just be `{task: {...}, user_id: userInput}`